### PR TITLE
[MS-824] Ensure sequential execution of UN20 tasks

### PR DESCRIPTION
**Pull request by meladRaouf**
This PR ensures sequential execution of UN20 tasks.

### CODE REVIEW

**Review by luhmirin-s — APPROVED** [review]

File: fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/screen/ObserveFingerprintScanStatusUseCase.kt [hunk 1]
```diff
@@ -63,7 +69,7 @@ class ObserveFingerprintScanStatusUseCase @Inject constructor(
         observeJob = null
     }
 
-    private suspend fun turnOnFlashingWhiteSmileLeds() {
+    private suspend fun turnOnFlashingWhiteSmileLeds()  {
```
Comment (luhmirin-s) [inline fingerprint/capture/src/main/java/com/simprints/fingerprint/capture/screen/ObserveFingerprintScanStatusUseCase.kt:72]: Accidental space

File: fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/v2/channel/MainMessageChannel.kt [hunk 2]
```diff
@@ -6,13 +6,31 @@ import com.simprints.fingerprint.infra.scanner.v2.incoming.main.MainMessageInput
...
-    incoming: MainMessageInputStream,
-    outgoing: MainMessageOutputStream
+    incoming: MainMessageInputStream, outgoing: MainMessageOutputStream
 ) : MessageChannel<MainMessageInputStream, MainMessageOutputStream>(incoming, outgoing) {
 
-    inline fun <reified R : IncomingMainMessage> sendMainModeCommandAndReceiveResponse(command: OutgoingMainMessage): Single<R> =
-        outgoing.sendMessage(command)
-            .doSimultaneously(incoming.receiveResponse())
+    // Mutex to ensure that only one thread can send and receive messages at a time
+    val mutex = Mutex()
```
Comment (luhmirin-s) [inline fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/v2/channel/MainMessageChannel.kt:20]: Should the same apply to other channels as well?

Comment (meladRaouf) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/v2/channel/MainMessageChannel.kt:20]: Yes, you're right; we should use the same logic for all channels. But the problem is that testing the OTA and other scanner flows will be hard, and there haven’t been any reports of concurrency issues with the other channels.

Comment (luhmirin-s) [reply fingerprint/infra/scanner/src/main/java/com/simprints/fingerprint/infra/scanner/v2/channel/MainMessageChannel.kt:20]: Maybe worth making a ticket to remember to look into it for upcoming major releases.

**Review by luhmirin-s — APPROVED** [review]

**Comment by meladRaouf** [comment]
I will close this PR for now, as it is difficult to validate whether it resolves the issue. We can revisit it later when we refactor how the scanner receives commands from SID.

**Pull request closed without merging**